hasura/migrations/default/1651000685281_alter_table_public_users_add_unique_address_circle_id_deleted_at/up.sql

```diff
@@ -0,0 +1 @@
+alter table "public"."users" add constraint "users_address_circle_id_deleted_at_key" unique ("address", "circle_id", "deleted_at");
```

Comment:
I tried to implement logic in createUser that "undeletes" soft-deleted users, and therefore this constraint should just require the `address` and `circle_id` columns. Is the logic not airtight enough to do that, or is existing production data preventing this? I think having a bunch of deleted users in a circle with the same address will make for some messy situations.

Reply:
yeah prior to your undelete feature there were cases of creating a new user after soft deleting one with the same address.The soft deleted users can't be hard deleted because of token_gifts fk constraints